avatica-shaded/pom.xml

```diff
@@ -130,6 +130,17 @@ language governing permissions and limitations under the License. -->
             <groupId>org.apache.httpcomponents</groupId>
             <artifactId>httpclient</artifactId>
             <version>${httpclient.version}</version>
+            <exclusions>
```

Comment:
is commons codec part of httpclient? 
This might not remove it and you might need to filter it as the three above (and then re-add it here)

Reply:
You're right. It's a transitive of httpclient

Reply:
Actually I take that back. 
`./org/apache/calcite/avatica/org/apache/commons/codec` is shaded too. 
So, needs to be filtered as above (including the meta-inf files) and to be added as an explicit dependency

Reply:
Yes, I see it's coming from `httpclient`:
```
mvn dependency:tree -Dverbose
...
[INFO] +- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
[INFO] |  +- commons-logging:commons-logging:jar:1.2:compile
[INFO] |  \- commons-codec:commons-codec:jar:1.15:compile
...
```
But that didn't work indeed.

Reply:
Actually, after looking at the content of the shaded JAR - it doesn't have the classes of this JAR, only maven's metadata:
```
~/workspace/dev/kafka-connect-storage-common/avatica-shaded/target> unzip kafka-connect-storage-common-avatica-shaded-10.1.0-SNAPSHOT.jar -d kafka-connect-storage-common-avatica-shaded-10.1.0-SNAPSHOT
...
find kafka-connect-storage-common-avatica-shaded-10.1.0-SNAPSHOT -name "commons-codec"
kafka-connect-storage-common-avatica-shaded-10.1.0-SNAPSHOT/META-INF/maven/commons-codec
kafka-connect-storage-common-avatica-shaded-10.1.0-SNAPSHOT/META-INF/maven/commons-codec/commons-codec
...
tree kafka-connect-storage-common-avatica-shaded-10.1.0-SNAPSHOT/META-INF/maven/commons-codec
kafka-connect-storage-common-avatica-shaded-10.1.0-SNAPSHOT/META-INF/maven/commons-codec
└── commons-codec
    ├── pom.properties
    └── pom.xml
```

So I just excluded the metadata from the shaded JAR.

Reply:
As we discussed offline, the files are there, shaded in the jar under:
`org/apache/calcite/avatica/org/apache/commons/codec`
We'll have to filter out those too